core/src/main/java/io/grpc/CallOptions.java

```diff
@@ -103,10 +118,12 @@ public CallOptions withCompression(@Nullable String compressorName) {
    * @param deadlineNanoTime the deadline in the clock as per {@link System#nanoTime()}.
    *                         {@code null} for unsetting the deadline.
    */
+  @Deprecated
   public CallOptions withDeadlineNanoTime(@Nullable Long deadlineNanoTime) {
-    CallOptions newOptions = new CallOptions(this);
-    newOptions.deadlineNanoTime = deadlineNanoTime;
-    return newOptions;
+    Deadline deadline = deadlineNanoTime != null
+        ? Deadline.after(deadlineNanoTime - System.nanoTime(), TimeUnit.NANOSECONDS)
```

Comment:
The other option is to change Deadline.timeRemaining() to return negative numbers. That would not result in nanos being equal, but it would be close. It does seem it could be useful to know how far in the past a deadline expired, which seems like an argument for returning negative numbers from timeRemaining(). The argument against returning negative numbers is applications may be buggy in their handling of it.